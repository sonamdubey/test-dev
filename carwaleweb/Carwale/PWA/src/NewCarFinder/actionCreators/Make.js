import 'isomorphic-fetch'

import {
	REQUEST_MAKE,
	RECEIVE_MAKE,
	MAKE_SELECTION,
	SET_MAKE_STORED_FILTERS
} from '../actionTypes'

import MakeApi from '../apis/Make'

import {
	getCityId
} from '../../selectors/LocationSelectors'

import {
	getMakeFilter,
	getBudgetFilter,
	getBodyTypeFilter,
	getFuelTypeFilter
} from '../selectors/NCFSelectors'

export const requestMake = () => {
	return {
		type: REQUEST_MAKE
	}
}

export const setMakeData = data => {
	return {
		type: RECEIVE_MAKE,
		data
	}
}

export const selectMake = id => {
	return {
		type: MAKE_SELECTION,
		id
	}
}

export const fetchMake = isFetching => (dispatch, getState) => {

	dispatch(requestMake(isFetching))
	const state = getState()
	const options = { ...getCityId(state), ...getBudgetFilter(state), ...getBodyTypeFilter(state), ...getFuelTypeFilter(state) }

	MakeApi
		.get(options)
		.then(data => {setTimeout(() => dispatch(setMakeData(data)), 300)})
		.catch(error => {
			//TODO: how should error affect the state ?
			console.log(error)
		})
}

export const setMakeStoredFilters = storedFilters => {
	return {
		type: SET_MAKE_STORED_FILTERS,
		storedFilters
	}
}
