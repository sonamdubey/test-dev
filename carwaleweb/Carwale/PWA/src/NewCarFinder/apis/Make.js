
import { MAKE_ENDPOINT } from '../constants'
import { serialzeObjectToQueryString } from '../../utils/Common'
/**
 * This returns a promise to fetch
 * fuel types availability for the selected filters
 * @return {Promise<>}
 */
const get = (options) => {
    return fetch(MAKE_ENDPOINT+"?CountOnly=true&"+serialzeObjectToQueryString(options))
    .then(response => {
        if (!response.ok) { throw response }
        return response.json()
    })
}
const MakeApi = {
    get
}
export default MakeApi
