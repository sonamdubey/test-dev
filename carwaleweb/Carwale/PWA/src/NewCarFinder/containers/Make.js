import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import {
	fetchMake,
	selectMake,
	setMakeStoredFilters
} from '../actionCreators/Make'

import {
	initToast,
	clearToast
} from '../../actionCreators/Toast'

import {
	setCurrentScreen
} from '../actionCreators/Filter'

import {
	makeCancelable
} from '../../utils/CancelablePromise'

import {
	getScreenId
} from '../selectors/NCFSelectors'

import MakeItem from '../components/MakeItem'

import { fireInteractiveTracking } from '../../utils/Analytics';

import SpeedometerLoader from '../../components/SpeedometerLoader'

import {
	createRipple
} from '../../utils/Ripple'
import {CATEGORY_NAME} from '../constants/index';
import { trackCustomData } from '../../utils/cwTrackingPwa';
class Make extends React.Component {
	constructor(props) {
		super(props)
	}

	componentDidMount() {
		const {
			fetchMake,
			setCurrentScreen,
			screenId
		} = this.props

		setCurrentScreen(screenId)
		fetchMake()
		trackCustomData(CATEGORY_NAME, "MakeScreenImpression", "NA", false)
	}

	componentWillUnmount() {
		this.props.clearToast()
	}

	componentDidUpdate(prevProps) {
		if(prevProps.isFetching && !this.props.isFetching)
		{
			this.props.setMakeStoredFilters(this.props.storedMakes)
		}
	}

	getPopularMake = (pop) => {
		const {
			makeData
		} = this.props

		let popularMakeData = makeData.popularMakes
		return popularMakeData
	}

	getOtherMake = () => {
		const {
			makeData
		} = this.props

		let otherMakeData = makeData.otherMakes
		return otherMakeData
	}

	getMakeList = (data, listType,section) => {
		let listItemIconType = ''

		if (listType === 'list-other-body__item') {
			listItemIconType = 'line'
		}
		let list = data.map(item => {
			const itemStatusClass = this.getItemStatus(item)
			return (
				<li
					key={item.makeId}
					className={"make-list__item " + listType + " " + itemStatusClass}
					onClick={this.handleMakeClick.bind(this, item,section)}
				>
					<MakeItem
						item={item}
					/>
				</li>
			)
		})

		return list
	}

	getItemStatus = (item) => {
		let status = ''

		if (item.isSelected) {
			status = 'active'
		}
		else if (!item.modelCount) {
			status = 'disable'
		}
		return status
	}

	handleMakeClick = (item,section,event) => {
		createRipple(event)
		if (item.modelCount) {
			this.props.selectMake(item.makeId)
		}
		else {
			this.setToast(item, event)
		}
		const status = (item.modelCount > 0 ? (item.isSelected ? "Deselect" : "Select") : "NotAvailableClick")
		fireInteractiveTracking(CATEGORY_NAME, "NCF_MakeFilter", status + "_" + item.makeName)
		trackCustomData(CATEGORY_NAME,"MakeFilterClick","section="+section+"|state="+status+"|make="+item.makeName,false)
	}

	setToast = (item, event) => {
		this.props.initToast({
			message: `There is no ${item.makeName} car matching your criteria.`,
			event: event
		})
	}

	render() {
		const {
			makeData,
			isFetching
		} = this.props

		if(isFetching){
			return <SpeedometerLoader />
		}
		let list,
		otherMakeList

		const isDataAvailable = (makeData.otherMakes != undefined)

		if (isDataAvailable) {
			let popularMakeData = this.getPopularMake()
			list = this.getMakeList(popularMakeData, "list-popular-make__item","PopularBrands")

			let otherMakeData = this.getOtherMake()
			otherMakeList = this.getMakeList(otherMakeData, "list-other-make__item","OtherBrands")
		}
		return (
			<div className="make-type-screen">
				<div className="screen__head">
					<p className="screen-head__title">Select your preferred Brands</p>
					<p className="screen-head__subtitle">Popular Brands</p>
					<ul className="make__list list__popular-make">
						{list}
					</ul>
				</div>
				<div className="screen__body">
					<p className="screen-body__title">Other Brands</p>
					<ul className="make__list list__other-make">
						{otherMakeList}
					</ul>
				</div>
			</div>
		)
	}
}

const mapStateToProps = (state) => {
	const {
		data: makeData,
		isFetching
	} = state.newCarFinder.make
	const {
		make : storedMakes
	} = state.filtersScreen.filters
	const screenId = getScreenId(state,'MakeFilter')
	return {
		makeData,
		isFetching,
		screenId,
		storedMakes
	}
}

const mapDispatchToProps = (dispatch) => {
	return {
		fetchMake: bindActionCreators(fetchMake, dispatch),
		selectMake: bindActionCreators(selectMake, dispatch),
		initToast: bindActionCreators(initToast, dispatch),
		clearToast: bindActionCreators(clearToast, dispatch),
		setCurrentScreen: bindActionCreators(setCurrentScreen, dispatch),
		setMakeStoredFilters: bindActionCreators(setMakeStoredFilters, dispatch)
	}
}

export default connect(mapStateToProps, mapDispatchToProps)(Make)
