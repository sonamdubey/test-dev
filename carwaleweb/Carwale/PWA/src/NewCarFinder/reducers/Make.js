import {
	REQUEST_MAKE,
	RECEIVE_MAKE,
	MAKE_SELECTION,
	SET_MAKE_STORED_FILTERS
} from '../actionTypes'

let initialMakeState = {
	isFetching: false,
	data: {
		popularMakes:[],
		otherMakes:[]
	}
}

export const make = (state = initialMakeState, action) => {
	switch (action.type) {
		case REQUEST_MAKE:
			return {
				...state,
				isFetching: true
			}

		case RECEIVE_MAKE:
		let popularSelectionState, otherSelectionState, popularMakes, otherMakes,makedata

		if(state.data.popularMakes!= undefined){
			popularSelectionState = state.data.popularMakes.reduce((acc, item) => {
				acc[item.makeId] = item
				return acc
			}, {})
		}
		if(state.data.otherMakes!= undefined){
			otherSelectionState = state.data.otherMakes.reduce((acc, item) => {
				acc[item.makeId] = item
				return acc
			}, {})
		}
		let selectionState = {...popularSelectionState, ...otherSelectionState}


		popularMakes= action.data.popularMakes.map( item => {
				item.isSelected = (selectionState[item.makeId] != undefined && selectionState[item.makeId].isSelected && item.modelCount > 0)
					return item
				})
		otherMakes= action.data.otherMakes.map( item => {
			item.isSelected = (selectionState[item.makeId] != undefined && selectionState[item.makeId].isSelected && item.modelCount > 0)
				return item
			})

        makedata = {popularMakes: popularMakes, otherMakes: otherMakes}
		return {
			...state,
			isFetching: false,
			data:makedata
		}

		case MAKE_SELECTION:
			const newData = state.data.popularMakes.map(item => {
				if (item.makeId === action.id) {
					return {
						...item,
						isSelected: !item.isSelected
					}
				}
				else {
					return item
				}
			})
			const otherNewData = state.data.otherMakes.map(item => {
				if (item.makeId === action.id) {
						return {
							...item,
							isSelected: !item.isSelected
						}
					}
					else {
						return item
					}
			})
			const concatedData = {popularMakes : newData, otherMakes : otherNewData}
			return {
				...state,
				data: concatedData
			}

		case SET_MAKE_STORED_FILTERS:
			const data = state.data.popularMakes.map(item => {
				if (action.storedFilters.indexOf(item.makeId) > -1 && item.modelCount > 0) {
					return {
						...item,
						isSelected: true
					}
				}
				else {
					return {
						...item,
						isSelected: false
					}
				}
			})
			const otherData = state.data.otherMakes.map(item => {
				if (action.storedFilters.indexOf(item.makeId) > -1 && item.modelCount > 0) {
						return {
							...item,
							isSelected: true
						}
					}
					else {
						return item
					}
			})
			const combinedData = {popularMakes : data, otherMakes : otherData}
			return {
				...state,
				data: combinedData
			}
		default:
			return state
	}
}
