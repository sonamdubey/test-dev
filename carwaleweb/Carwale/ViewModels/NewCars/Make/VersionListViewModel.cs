using Carwale.DTOs.NewCars;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace Carwale.UI.ViewModels.NewCars.Make
{
    public class VersionListViewModel
    {
        public string MakeName { get; set; }
        public List<NewCarVersionsDTO> Versions { get; set; }
    }
}