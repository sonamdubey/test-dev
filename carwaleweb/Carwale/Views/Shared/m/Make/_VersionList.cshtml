@model Carwale.UI.ViewModels.NewCars.Make.VersionListViewModel

@if(Model != null && Model.Versions.Count > 0)
{
    <ul class="card-wrapper version-list__content">
        @foreach(var version in Model.Versions)
        {
            <li class="card-wrapper__item version-list__item">
                <div data-version="@version.Id" class="version-list__item-content">
                    <div class="version__container">
                        <div class="version__card">
                            <div class="version__content">
                                <a href="@Carwale.Utility.ManageCarUrl.CreateVersionUrl(Model.MakeName,version.MaskingName,version.VersionMaskingName,true)" title="" class="version__box">
                                    <h3><span class="version__name">@version.Version</span></h3>
                                    <ul class="version__feature-list">
                                        <li class="feature-list__item">@version.MaxPower</li>
                                        <li class="feature-list__item">@version.Displacement</li>
                                        <li class="feature-list__item">@version.TransmissionType</li>
                                        <li class="feature-list__item"> @version.CarFuelType </li>
                                    </ul>
                                </a>
                                @if (version.CarPriceOverview?.Price != null)
                                {
                                    string priceLable = version.CarPriceOverview.PriceStatus == (int)Carwale.Entity.PriceBucket.HaveUserCity ? (version.CarPriceOverview.PriceLabel?.Replace(" (incl. GST, Estimated)", string.Empty)?.Replace(" (incl. GST)", string.Empty) + ", " + version.CarPriceOverview?.City?.CityName ?? string.Empty) : version.CarPriceOverview.PriceLabel;
                                    <div class="on-road-price__content">
                                        <div class="price__content">
                                            <div class="price__column">
                                                <span class="price-content__amount">₹ @Carwale.UI.Common.CommonOpn.FormatNumeric(version.CarPriceOverview.Price.ToString())</span>
                                                <p class="price-content__title">@priceLable</p>
                                            </div>
                                            <div class="price__emi-content text--italic">
                                                <p class="price-content__title">Emi</p>
                                                <span class="price-content__emi-amount">₹ @Carwale.BL.Calculation.Calculation.CalculateEmi(version.CarPriceOverview.Price)</span>
                                            </div>
                                        </div>
                                    </div>
                                }                                
                            </div>
                        </div>
                    </div>
                </div>
            </li>
        }        
    </ul>
}

