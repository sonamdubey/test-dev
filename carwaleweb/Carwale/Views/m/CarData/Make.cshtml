@model Carwale.DTOs.NewCars.MakePageDTO_Mobile
@using Carwale.Entity.Enum;
@using Carwale.UI.Common;
@using System.Linq;
@using Carwale.BL.Experiments;
@{
    if (Model.MakeDetails != null)
    {
        ViewBag.DefineAdSlots =
            @"<script type='text/javascript'>
    var AdSlots={};
        googletag.cmd.push(function () {
        googletag.pubads().disableInitialLoad();
        googletag.pubads().setCentering(true);
        AdSlots['div-gpt-ad-1419227721763-0']=googletag.defineSlot('/1017752/" + ViewBag.adPath + @"CarWale_Mobile_ROS_320x50', [[320, 50], [320, 100], [320, 150]], 'div-gpt-ad-1419227721763-0').addService(googletag.pubads()).setCollapseEmptyDiv(true);
        //Ad for quatation page bottom Ad
        AdSlots['div-gpt-ad-1435297201507-0']=googletag.defineSlot('/1017752/" + ViewBag.adPath + @"CarWale_Mobile_ROS_300x250', [300, 250], 'div-gpt-ad-1435297201507-0').addService(googletag.pubads());
        googletag.pubads().setTargeting('Make', '" + Model.MakeDetails.MakeName + @"');
        googletag.pubads().setTargeting('Section', '');
        googletag.pubads().setTargeting('City', '" + MobileWeb.Common.CookiesCustomers.MasterCity + @"');
        googletag.pubads().collapseEmptyDivs(true);
        googletag.pubads().enableSingleRequest();
        googletag.enableServices();
    });
</script>";
    }
    Layout = "~/views/shared/m/_MLayoutV3.cshtml";
    string label = "makeid=" + Model.MakeDetails.MakeId + "|source=43";
}
@section Plugins{
    <script type="text/javascript" src="/static/m/js/plugins/swiper.js" defer></script>
    <script type="text/javascript" src="/static/m/js/plugins/beauty-tooltips.js" defer></script>
    <script type="text/javascript" src="/static/m/js/plugins/flip.js" defer></script>
}
@section atfcss{
    <link rel="stylesheet" href="/static/sass/partials/make.css" type="text/css" inline>
    <link href="/static/sass/partials/review-contest-slug.css" rel="stylesheet" inline>
}
@section btfcss{
    <link rel="stylesheet" href="/static/sass/m/css/globalcitypopup.css" type="text/css">
    <link rel="stylesheet" href="/static/sass/m/css/globalcarsearch.css" type="text/css">
    <link rel="stylesheet" href="/static/sass/m/css/advantagecitypopup.css" type="text/css">
    <link rel="stylesheet" href="/static/sass/m/css/nav.css" type="text/css">
    <link rel="stylesheet" href="/static/sass/m/css/selectcitypq.css" type="text/css">
    <link rel="stylesheet" href="/static/sass/m/css/footer.css" type="text/css">
    <link rel="stylesheet" href="/static/sass/m/css/swiper.css" type="text/css">
    <link rel="stylesheet" href="/static/sass/partials/model-image-carousel.css" type="text/css">
    <link href="/static/sass/partials/model-videos-carousel.css" rel="stylesheet">
}
@section MetaData{
    @{Html.RenderAction("MetaData", "ModelComponent", Model.MetaData);}
}
@if (Model.MakeDetails != null)
{
    @section Scripts{
        <script>
            var makeId = @Model.MakeDetails.MakeId;
        </script>
        <script type="text/javascript" src="/static/m/js/new-car-make.js" defer></script>
        <script type="text/javascript" src="/static/m/js/upcoming-form.js" defer></script>
        @if (Model.MakeDetails.MakeId.Equals(16))
        {
            <script type="text/javascript" src="/static/src/jquery.obj.min.js" defer></script>
        }
        <script type="text/javascript">var product = {category: ['new', '@UrlRewrite.FormatSpecial(Model.MakeDetails.MakeName)-cars']};</script>
    }
}
<div class="clear"></div>
<div data-cwticat="ReviewLandingLinkage" data-cwtiact="MakePageSlugImpression" data-cwtilbl="@label" data-cwtccat="ReviewLandingLinkage" data-cwtcact="MakePageSlugClick" data-cwtclbl="@label" cwt>@Html.Partial("~/Views/Shared/m/UserReviews/_UserReviewSlug.cshtml")</div>

<div id="divAdBar">
    <div id='div-gpt-ad-1419227721763-0' style='width:320px; margin:20px auto 10px auto;' data-load-immediate="true">
        <script type='text/javascript'>
            googletag.cmd.push(function () { googletag.display('div-gpt-ad-1419227721763-0'); });
        </script>
    </div>
</div>
<div class="clear"></div>

<section class="container">
    <div id="main-container">
        @if (Model.MakeDetails != null)
        {
        <div>
            <div class="make-car-container m-special-skin-text m-no-bg-color">
                <h1 class="make-heading  m-no-bg-color">@Model.MakeDetails.MakeName Cars</h1>
            </div>
            @if (Model.NewCarModels.Count > 0)
            {
                @Html.Partial("m/CarData/_ModelList", Model)
                <div id="toastBox" class="toast-box hide"></div>
            }
            @{Html.RenderAction("Description", "ModelComponent", new { makeId = Model.MakeDetails.MakeId, makeName = Model.MakeDetails.MakeName });}

            <div class="pgsubhead margin-left10 margin-right10 margin-bottom10 margin-top20">
                <h2 class="special-skin-text m-special-skin-text text-bold">Other popular Brands</h2>
            </div>
            <div class="box content-inner-block-10 rounded-corner2 make-popular-brands">
                @{Html.RenderAction("AllMakesWidget", "CarWidgets", new { excludeMakeId = Model.MakeDetails.MakeId, isMobile = true });}
            </div>
            <div class="box content-inner-block-10 car-carousel">
                @{Html.RenderAction("TopLatestNewsForMake", "ModelComponent", new { makeId = Model.MakeDetails.MakeId, makeName = Model.MakeDetails.MakeName });}
            </div>

            @if (Model.Images != null && Model.Images.ModelPhotos != null && Model.Images.ModelPhotos.Count > 0)
            {
                <div class="box content-inner-block-10 car-carousel">
                    @{Html.RenderPartial("~/Views/Shared/m/CarData/_ModelImages.cshtml", Model.Images);}
                </div>
            }
            @if (Model.MakeVideos != null && Model.MakeVideos.Count > 0)
            {
                <div class="box content-inner-block-10 car-carousel">
                    @Html.Partial("m/CarData/_Videos", Tuple.Create(Model.MakeVideos, ContentPages.MakePage))
                </div>
            }
        </div>
        }        
    </div>
    <div class="clear"></div>    
    <div class="banner-div">
        <div id="divBottomAdBar" class="ad-div">
            <div id='div-gpt-ad-1435297201507-0' style='height:250px; width:300px;' class="text-center margin-bottom20">
                <script type='text/javascript'>googletag.cmd.push(function () { googletag.display('div-gpt-ad-1435297201507-0'); });</script>
            </div>
        </div>
    </div>

    <div class="margin-left10" id="br-cr">
        <ul class="breadcrumb m-special-skin-text">
            <li>
                <a href="/m/" class="normal">Home</a>  &rsaquo;
            </li>
            <li>
                <span class="lightgray"> @Model.MakeDetails.MakeName Cars</span>
            </li>
        </ul>
    </div>
</section>
