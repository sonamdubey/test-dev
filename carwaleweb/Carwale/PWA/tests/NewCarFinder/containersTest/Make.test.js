import React from 'react'
import { Redirect, MemoryRouter } from 'react-router-dom'

import Make from '../../../src/NewCarFinder/containers/Make'

import {
	SET_TOAST
} from '../../../src/actionTypes'

import {
	SET_CURRENT_SCREEN,
	REQUEST_MAKE,
	RECEIVE_MAKE,
	MAKE_SELECTION,
} from '../../../src/NewCarFinder/actionTypes'


const makeData = {
"popularMakes": [
    {"makeId": 50,"makeName": "Force Motors","modelCount": 2}
],
"otherMakes": [
    {"makeId": 10,"makeName": "Maruti Suzuki","modelCount": 0},
    {"makeId": 7,"makeName": "Honda","modelCount": 0},
    {"makeId": 17,"makeName": "Toyota","modelCount": 0}
]
}

jest.mock('../../src/NewCarFinder/apis/Make', () => {
    const makeData = {
        "popularMakes": [
            {"makeId": 50,"makeName": "Force Motors","modelCount": 2}
        ],
        "otherMakes": [
            {"makeId": 10,"makeName": "Maruti Suzuki","modelCount": 0},
            {"makeId": 7,"makeName": "Honda","modelCount": 0},
            {"makeId": 17,"makeName": "Toyota","modelCount": 0}
        ]
        }
    return {
        get: jest.fn((options) => Promise.resolve(makeData))
    }
})

jest.mock('../../../src/utils/Ripple')


describe("Tests for Make container component", () => {

    let props
    let store
    let enzymeWrapper
    let state
    const Wrapper = () => {
        if (!enzymeWrapper) {
            store = mockStore(state)
            enzymeWrapper = shallowWithStore(<Make {...props} />, store).dive()
        }
        return enzymeWrapper
    }

    beforeEach(() => {
        props = undefined
        state = getMockInitialStore()
        enzymeWrapper = undefined
        store = undefined
    })

    describe("when no make data is available", () => {
        it("should match snapshot", () => {
            const wrapper = Wrapper()
            expect(wrapper).toMatchSnapshot()
        })
    })

    describe("when component is mounted", () => {
        it("should dispatch appropriate actions", (done) => {
            /**
             * documentation for done and use case : https://jasmine.github.io/2.5/introduction.html#section-58
             * and https://stackoverflow.com/questions/47275706/async-function-not-behaving-as-i-expect-with-jest
             */
            const wrapper = Wrapper()
            const expectedActions = store.getActions()
            expect(expectedActions).toHaveLength(2)

            expect(expectedActions).toContainEqual({
                type: SET_CURRENT_SCREEN,
                screenId: 3
            })

            expect(expectedActions).toContainEqual({
                type: REQUEST_MAKE
            })

            // http://www.bambielli.com/til/2017-09-19-process-nexttick/
            // Please read this blog before attempting a change : http://voidcanvas.com/setimmediate-vs-nexttick-vs-settimeout/
            process.nextTick(() => {
                setTimeout(() => {
                    try {
                        expect(store.getActions()).toHaveLength(3)
                        expect(store.getActions()).toContainEqual({
                            type: RECEIVE_MAKE,
                            data: makeData
                        })
                    } catch (e) {
                        return done(e);
                    }
                    done();
                },300)
            });
        })
    })

    describe("when data is available", () => {
        it("should match snapshot", () => {
            state.newCarFinder.make.data = makeData
            const wrapper = Wrapper()
            expect(wrapper).toMatchSnapshot()
        })
    })

    describe("when MAKE with count > 0 is clicked", () => {
        it("should dispatch Make_SELECTION action", () => {
            state.newCarFinder.make.data = makeData
            const wrapper = Wrapper()
            let activeItems = makeData.popularMakes.filter(x=> x.isSelected)
            activeItems.concat( makeData.otherMakes.filter(x=> x.isSelected))
           // let activeItems = makeData.filter(x => x.carCount > 0)
            if (activeItems.length) {
                const firstItem = wrapper.find('.make-list__item').findWhere(x => x.key() == activeItems[0].makeId)
                firstItem.simulate('click')
                const expectedActions = store.getActions()
                expect(expectedActions).toContainEqual({
                    type: MAKE_SELECTION,
                    id: parseInt(firstItem.key())
                })
            }
        })
     })

    describe("when make with modelcount == 0 is clicked", () => {
        it("should not dispatch MAKE_SELECTION action", () => {
            state.newCarFinder.make.data = makeData
            const wrapper = Wrapper()
            let disabledItems = makeData.popularMakes.filter(x => x.isSelected == 0)
            disabledItems.concat(makeData.otherMakes.filter(x => x.isSelected == 0))
            if (disabledItems.length) {
                const firstItem = wrapper.find('.make-list__item').findWhere(x => x.key() == disabledItems[0].makeId)
                firstItem.simulate('click')
                const expectedActions = store.getActions().filter(x => x.type == SET_TOAST)
                expect(expectedActions.length).toBe(1)
                //expect(expectedActions[0].message).toBe(``)
            }
        })
    })

})
