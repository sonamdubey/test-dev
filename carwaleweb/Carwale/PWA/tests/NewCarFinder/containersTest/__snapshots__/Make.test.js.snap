// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Make container component when data is available should match snapshot 1`] = `
<div
  className="make-type-screen"
>
  <div
    className="screen__head"
  >
    <p
      className="screen-head__title"
    >
      Select your preferred Brands
    </p>
    <p
      className="screen-head__subtitle"
    >
      Popular Brands
    </p>
    <ul
      className="make__list list__popular-make"
    >
      <li
        className="make-list__item list-popular-make__item "
        key="50"
        onClick={[Function]}
      >
        <MakeItem
          item={
            Object {
              "makeId": 50,
              "makeName": "Force Motors",
              "modelCount": 2,
            }
          }
          showCount={true}
        />
      </li>
    </ul>
  </div>
  <div
    className="screen__body"
  >
    <p
      className="screen-body__title"
    >
      Other Brands
    </p>
    <ul
      className="make__list list__other-make"
    >
      <li
        className="make-list__item list-other-make__item disable"
        key="10"
        onClick={[Function]}
      >
        <MakeItem
          item={
            Object {
              "makeId": 10,
              "makeName": "Maruti Suzuki",
              "modelCount": 0,
            }
          }
          showCount={true}
        />
      </li>
      <li
        className="make-list__item list-other-make__item disable"
        key="7"
        onClick={[Function]}
      >
        <MakeItem
          item={
            Object {
              "makeId": 7,
              "makeName": "Honda",
              "modelCount": 0,
            }
          }
          showCount={true}
        />
      </li>
      <li
        className="make-list__item list-other-make__item disable"
        key="17"
        onClick={[Function]}
      >
        <MakeItem
          item={
            Object {
              "makeId": 17,
              "makeName": "Toyota",
              "modelCount": 0,
            }
          }
          showCount={true}
        />
      </li>
    </ul>
  </div>
</div>
`;

exports[`Tests for Make container component when no make data is available should match snapshot 1`] = `
<div
  className="make-type-screen"
>
  <div
    className="screen__head"
  >
    <p
      className="screen-head__title"
    >
      Select your preferred Brands
    </p>
    <p
      className="screen-head__subtitle"
    >
      Popular Brands
    </p>
    <ul
      className="make__list list__popular-make"
    />
  </div>
  <div
    className="screen__body"
  >
    <p
      className="screen-body__title"
    >
      Other Brands
    </p>
    <ul
      className="make__list list__other-make"
    />
  </div>
</div>
`;
